relay-server/src/actors/outcome.rs

```diff
@@ -475,24 +479,40 @@ pub enum OutcomeError {
 struct HttpOutcomeProducer {
     config: Arc<Config>,
     unsent_outcomes: Vec<TrackRawOutcome>,
-    pending_flush_handle: Option<SpawnHandle>,
+    // Sleep must be `Box<Pin>` in order for it to be Unpin.
+    // See: https://docs.rs/tokio/1.21.0/tokio/time/struct.Sleep.html#examples
+    pending_flush_handle: Either<future::Pending<()>, Pin<Box<tokio::time::Sleep>>>,
```

Comment:
Lol, pretty sure getting rid of `Either` hacks is one benefit of migrating to modern futures.  Oh well.